rerun_py/src/catalog/catalog_client.rs

```diff
@@ -50,13 +51,13 @@ impl PyCatalogClient {
         entry_details
             .into_iter()
             .map(|entry| {
-                let entry_id = Py::new(
-                    py,
+                let entry_id = EntryId::try_from(
                     entry
                         .id
-                        .ok_or(PyRuntimeError::new_err("No id in entry"))
-                        .and_then(PyEntryId::try_from)?,
-                )?;
+                        .ok_or(MissingGrpcFieldError::new_err("No id in entry"))?,
+                )
+                .map_err(|err| PyValueError::new_err(err.to_string()))?;
+                let entry_id = Py::new(py, PyEntryId::from(entry_id))?;
```

Comment:
That's awful, but then again that's only awful because we leaked the raw protobuf EntryDetails all the way here